BART Summarization max tokens?

### Is there an existing issue for this?

- [X] I have searched the existing issues and did not find a match.


### What are you working on?

I am trying to summarize potentially long texts with distilbart_xsum_12_6.

### Current Behavior

Currently I get an error on long texts: 

```
TFInvalidArgumentException: {{function_node __inference_pruned_56247}} {{function_node __inference_pruned_56247}} indices[1054] = 1056 is not in [0, 1026)
```

(full error at the bottom)

### Expected Behavior

Maybe not *expected* behavior, but I always assumed something was going on under the hood to handle texts longer than a particular model's max context length.  Does no such mechanism exist for BART, or in general?

### Steps To Reproduce

A rough example:

```
from sparknlp.base import *
from sparknlp.annotator import *

documentAssembler = DocumentAssembler() \
    .setInputCol('text') \
    .setOutputCol('document')

bart = BartTransformer.pretrained("distilbart_xsum_12_6") \
            .setTask("summarize:") \
            .setMaxOutputLength(250) \
            .setInputCols(["document"]) \
            .setOutputCol("summary")

nlp_pipeline = Pipeline(stages=[
    documentAssembler, 
    bart
])

pipeline_model = nlp_pipeline.fit(spark.createDataFrame([['']]).toDF('text'))

sentences = [
  [" ".join(["word"]*1027)]
  for i in range(3)
]

data = spark.createDataFrame(sentences).toDF("text")
data.show()

display(pipeline_model.transform(data))
```

### Spark NLP version and Apache Spark

sparknlp 4.4.0
spark 3.3.2

****

```
org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 104.0 failed 4 times, most recent failure: Lost task 1.3 in stage 104.0 (TID 1844) (10.139.64.7 executor 15): org.tensorflow.exceptions.TFInvalidArgumentException: {{function_node __inference_pruned_56247}} {{function_node __inference_pruned_56247}} indices[1054] = 1056 is not in [0, 1026)
	 [[{{node encoder/embed_positions/embedding_lookup}}]]
	 [[StatefulPartitionedCall_1/StatefulPartitionedCall/StatefulPartitionedCall]]
	at org.tensorflow.internal.c_api.AbstractTF_Status.throwExceptionIfNotOK(AbstractTF_Status.java:87)
	at org.tensorflow.Session.run(Session.java:850)
	at org.tensorflow.Session.access$300(Session.java:82)
	at org.tensorflow.Session$Runner.runHelper(Session.java:552)
	at org.tensorflow.Session$Runner.runNoInit(Session.java:499)
	at org.tensorflow.Session$Runner.run(Session.java:495)
	at com.johnsnowlabs.ml.ai.Bart.tag(Bart.scala:169)
	at com.johnsnowlabs.ml.ai.Bart.$anonfun$predict$1(Bart.scala:763)
	at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:293)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
	at scala.collection.TraversableLike.flatMap(TraversableLike.scala:293)
	at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:290)
	at scala.collection.mutable.ArrayOps$ofRef.flatMap(ArrayOps.scala:198)
	at com.johnsnowlabs.ml.ai.Bart.predict(Bart.scala:749)
	at com.johnsnowlabs.nlp.annotators.seq2seq.BartTransformer.batchAnnotate(BartTransformer.scala:487)
	at com.johnsnowlabs.nlp.HasBatchedAnnotate.$anonfun$batchProcess$1(HasBatchedAnnotate.scala:59)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:486)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:492)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:761)
	at org.apache.spark.sql.execution.collect.UnsafeRowBatchUtils$.encodeUnsafeRows(UnsafeRowBatchUtils.scala:82)
	at org.apache.spark.sql.execution.collect.Collector.$anonfun$processFunc$1(Collector.scala:208)
	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$3(ResultTask.scala:75)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$1(ResultTask.scala:75)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:55)
	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:174)
	at org.apache.spark.scheduler.Task.$anonfun$run$5(Task.scala:142)
	at com.databricks.unity.EmptyHandle$.runWithAndClose(UCSHandle.scala:125)
	at org.apache.spark.scheduler.Task.$anonfun$run$1(Task.scala:142)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.scheduler.Task.run(Task.scala:97)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$13(Executor.scala:904)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1713)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:907)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:761)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:3377)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:3309)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:3300)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:3300)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1429)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1429)
	at scala.Option.foreach(Option.scala:407)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1429)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3589)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:3527)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:3515)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:51)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$runJob$1(DAGScheduler.scala:1178)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.databricks.spark.util.FrameProfiler$.record(FrameProfiler.scala:80)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:1166)
	at org.apache.spark.SparkContext.runJobInternal(SparkContext.scala:2737)
	at org.apache.spark.sql.execution.collect.Collector.$anonfun$runSparkJobs$1(Collector.scala:349)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.databricks.spark.util.FrameProfiler$.record(FrameProfiler.scala:80)
	at org.apache.spark.sql.execution.collect.Collector.runSparkJobs(Collector.scala:293)
	at org.apache.spark.sql.execution.collect.Collector.collect(Collector.scala:377)
	at org.apache.spark.sql.execution.collect.Collector$.collect(Collector.scala:128)
	at org.apache.spark.sql.execution.collect.Collector$.collect(Collector.scala:135)
	at org.apache.spark.sql.execution.qrc.InternalRowFormat$.collect(cachedSparkResults.scala:123)
	at org.apache.spark.sql.execution.qrc.InternalRowFormat$.collect(cachedSparkResults.scala:111)
	at org.apache.spark.sql.execution.qrc.InternalRowFormat$.collect(cachedSparkResults.scala:93)
	at org.apache.spark.sql.execution.qrc.ResultCacheManager.$anonfun$computeResult$1(ResultCacheManager.scala:537)
	at com.databricks.spark.util.FrameProfiler$.record(FrameProfiler.scala:80)
	at org.apache.spark.sql.execution.qrc.ResultCacheManager.collectResult$1(ResultCacheManager.scala:529)
	at org.apache.spark.sql.execution.qrc.ResultCacheManager.computeResult(ResultCacheManager.scala:549)
	at org.apache.spark.sql.execution.qrc.ResultCacheManager.$anonfun$getOrComputeResultInternal$1(ResultCacheManager.scala:402)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.execution.qrc.ResultCacheManager.getOrComputeResultInternal(ResultCacheManager.scala:395)
	at org.apache.spark.sql.execution.qrc.ResultCacheManager.getOrComputeResult(ResultCacheManager.scala:289)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$executeCollectResult$1(SparkPlan.scala:506)
	at com.databricks.spark.util.FrameProfiler$.record(FrameProfiler.scala:80)
	at org.apache.spark.sql.execution.SparkPlan.executeCollectResult(SparkPlan.scala:503)
	at org.apache.spark.sql.Dataset.collectResult(Dataset.scala:3453)
	at org.apache.spark.sql.Dataset.$anonfun$collectResult$1(Dataset.scala:3444)
	at org.apache.spark.sql.Dataset.$anonfun$withAction$3(Dataset.scala:4368)
	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:809)
	at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4366)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$8(SQLExecution.scala:227)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:410)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$1(SQLExecution.scala:172)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:1038)
	at org.apache.spark.sql.execution.SQLExecution$.withCustomExecutionEnv(SQLExecution.scala:122)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:360)
	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4366)
	at org.apache.spark.sql.Dataset.collectResult(Dataset.scala:3443)
	at com.databricks.backend.daemon.driver.OutputAggregator$.withOutputAggregation0(OutputAggregator.scala:267)
	at com.databricks.backend.daemon.driver.OutputAggregator$.withOutputAggregation(OutputAggregator.scala:101)
	at com.databricks.backend.daemon.driver.PythonDriverLocalBase.generateTableResult(PythonDriverLocalBase.scala:723)
	at com.databricks.backend.daemon.driver.JupyterDriverLocal.computeListResultsItem(JupyterDriverLocal.scala:839)
	at com.databricks.backend.daemon.driver.JupyterDriverLocal$JupyterEntryPoint.addCustomDisplayData(JupyterDriverLocal.scala:258)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)
	at py4j.Gateway.invoke(Gateway.java:306)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:195)
	at py4j.ClientServerConnection.run(ClientServerConnection.java:115)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.tensorflow.exceptions.TFInvalidArgumentException: {{function_node __inference_pruned_56247}} {{function_node __inference_pruned_56247}} indices[1054] = 1056 is not in [0, 1026)
	 [[{{node encoder/embed_positions/embedding_lookup}}]]
	 [[StatefulPartitionedCall_1/StatefulPartitionedCall/StatefulPartitionedCall]]
	at org.tensorflow.internal.c_api.AbstractTF_Status.throwExceptionIfNotOK(AbstractTF_Status.java:87)
	at org.tensorflow.Session.run(Session.java:850)
	at org.tensorflow.Session.access$300(Session.java:82)
	at org.tensorflow.Session$Runner.runHelper(Session.java:552)
	at org.tensorflow.Session$Runner.runNoInit(Session.java:499)
	at org.tensorflow.Session$Runner.run(Session.java:495)
	at com.johnsnowlabs.ml.ai.Bart.tag(Bart.scala:169)
	at com.johnsnowlabs.ml.ai.Bart.$anonfun$predict$1(Bart.scala:763)
	at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:293)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
	at scala.collection.TraversableLike.flatMap(TraversableLike.scala:293)
	at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:290)
	at scala.collection.mutable.ArrayOps$ofRef.flatMap(ArrayOps.scala:198)
	at com.johnsnowlabs.ml.ai.Bart.predict(Bart.scala:749)
	at com.johnsnowlabs.nlp.annotators.seq2seq.BartTransformer.batchAnnotate(BartTransformer.scala:487)
	at com.johnsnowlabs.nlp.HasBatchedAnnotate.$anonfun$batchProcess$1(HasBatchedAnnotate.scala:59)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:486)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:492)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:761)
	at org.apache.spark.sql.execution.collect.UnsafeRowBatchUtils$.encodeUnsafeRows(UnsafeRowBatchUtils.scala:82)
	at org.apache.spark.sql.execution.collect.Collector.$anonfun$processFunc$1(Collector.scala:208)
	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$3(ResultTask.scala:75)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.scheduler.ResultTask.$anonfun$runTask$1(ResultTask.scala:75)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:55)
	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:174)
	at org.apache.spark.scheduler.Task.$anonfun$run$5(Task.scala:142)
	at com.databricks.unity.EmptyHandle$.runWithAndClose(UCSHandle.scala:125)
	at org.apache.spark.scheduler.Task.$anonfun$run$1(Task.scala:142)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.scheduler.Task.run(Task.scala:97)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$13(Executor.scala:904)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1713)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:907)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:761)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
```

--- a/python/test/annotator/seq2seq/bart_transformer_test.py
+++ b/python/test/annotator/seq2seq/bart_transformer_test.py
@@ -46,7 +46,7 @@ class BartTransformerQATestSpec(unittest.TestCase):
 
         results.select("documents.result", "answers.result").show(truncate=False)
 
-@pytest.mark.fast
+@pytest.mark.slow
 class BartTransformerMaxLengthTestSpec(unittest.TestCase):
     def setUp(self):
         self.spark = SparkContextForTest.spark

--- a/python/test/annotator/seq2seq/bart_transformer_test.py
+++ b/python/test/annotator/seq2seq/bart_transformer_test.py
@@ -46,6 +46,81 @@ class BartTransformerQATestSpec(unittest.TestCase):
 
         results.select("documents.result", "answers.result").show(truncate=False)
 
+@pytest.mark.fast
+class BartTransformerMaxLengthTestSpec(unittest.TestCase):
+    def setUp(self):
+        self.spark = SparkContextForTest.spark
+
+    def runTest(self):
+        data = self.spark.createDataFrame([
+            [1, """
+            Heat oven to 200C/180C fan/gas 6. Line each hole of a 12-hole muffin tin with a thin strip of baking 
+            parchment across the middle that’s long enough so the ends stick out a centimetre or two – use a dab of
+             butter to stick in place. Roll out two thirds of the pastry on a lightly floured surface and stamp out 
+             12 x 10cm circles (you may need to re-roll trimmings). Press a circle into each hole to line.
+             
+            Sprinkle 1 tsp of breadcrumbs into the base of each pie. Tip the rest of the crumbs into a mixing bowl. 
+            Squeeze in the sausage meat, discarding the skins, along with the bacon, mace, pepper, sage and just a 
+            little salt. Get your hands in and mash and squish everything together until the breadcrumbs have just 
+            about disappeared. Divide mixture between the holes, packing in firmly and shaping to a dome in the middle.
+             
+            Roll out the remaining pastry and stamp out 12 x 7cm circles. Brush with a little egg and add a top to 
+            each pie, egg-side down to stick, carefully pressing pastry edges together to seal. Brush with more egg 
+            (don’t throw away leftovers) and sprinkle with sesame seeds. Bake for 30 mins until golden then carefully 
+            remove the pies from the tin, using the parchment ends to help you lift them out. Sit on a parchment lined 
+            baking tray, brush all round the sides with more egg and put back in the oven for 8 mins. Cool completely 
+            then eat with piccalilli, or your favourite pickle.             
+            
+            Heat oven to 200C/180C fan/gas 6. Line each hole of a 12-hole muffin tin with a thin strip of baking 
+            parchment across the middle that’s long enough so the ends stick out a centimetre or two – use a dab of
+             butter to stick in place. Roll out two thirds of the pastry on a lightly floured surface and stamp out 
+             12 x 10cm circles (you may need to re-roll trimmings). Press a circle into each hole to line.
+             
+            Sprinkle 1 tsp of breadcrumbs into the base of each pie. Tip the rest of the crumbs into a mixing bowl. 
+            Squeeze in the sausage meat, discarding the skins, along with the bacon, mace, pepper, sage and just a 
+            little salt. Get your hands in and mash and squish everything together until the breadcrumbs have just 
+            about disappeared. Divide mixture between the holes, packing in firmly and shaping to a dome in the middle.
+             
+            Roll out the remaining pastry and stamp out 12 x 7cm circles. Brush with a little egg and add a top to 
+            each pie, egg-side down to stick, carefully pressing pastry edges together to seal. Brush with more egg 
+            (don’t throw away leftovers) and sprinkle with sesame seeds. Bake for 30 mins until golden then carefully 
+            remove the pies from the tin, using the parchment ends to help you lift them out. Sit on a parchment lined 
+            baking tray, brush all round the sides with more egg and put back in the oven for 8 mins. Cool completely 
+            then eat with piccalilli, or your favourite pickle.      
+            
+            Heat oven to 200C/180C fan/gas 6. Line each hole of a 12-hole muffin tin with a thin strip of baking 
+            parchment across the middle that’s long enough so the ends stick out a centimetre or two – use a dab of
+             butter to stick in place. Roll out two thirds of the pastry on a lightly floured surface and stamp out 
+             12 x 10cm circles (you may need to re-roll trimmings). Press a circle into each hole to line.
+             
+            Sprinkle 1 tsp of breadcrumbs into the base of each pie. Tip the rest of the crumbs into a mixing bowl. 
+            Squeeze in the sausage meat, discarding the skins, along with the bacon, mace, pepper, sage and just a 
+            little salt. Get your hands in and mash and squish everything together until the breadcrumbs have just 
+            about disappeared. Divide mixture between the holes, packing in firmly and shaping to a dome in the middle.
+             
+            Roll out the remaining pastry and stamp out 12 x 7cm circles. Brush with a little egg and add a top to 
+            each pie, egg-side down to stick, carefully pressing pastry edges together to seal. Brush with more egg 
+            (don’t throw away leftovers) and sprinkle with sesame seeds. Bake for 30 mins until golden then carefully 
+            remove the pies from the tin, using the parchment ends to help you lift them out. Sit on a parchment lined 
+            baking tray, brush all round the sides with more egg and put back in the oven for 8 mins. Cool completely 
+            then eat with piccalilli, or your favourite pickle.      
+            """.strip().replace("\n", " ")]]).toDF("id", "text")
+
+        document_assembler = DocumentAssembler() \
+            .setInputCol("text") \
+            .setOutputCol("documents")
+
+        bart = BartTransformer.pretrained("distilbart_xsum_12_6") \
+            .setTask("summarize:") \
+            .setMaxOutputLength(30) \
+            .setInputCols(["documents"]) \
+            .setOutputCol("summaries")
+
+        pipeline = Pipeline().setStages([document_assembler, bart])
+        results = pipeline.fit(data).transform(data)
+
+        results.select("summaries.result").show(truncate=False)
+
 
 @pytest.mark.slow
 class BartTransformerSummaryTestSpec(unittest.TestCase):

--- a/src/main/scala/com/johnsnowlabs/ml/ai/Bart.scala
+++ b/src/main/scala/com/johnsnowlabs/ml/ai/Bart.scala
@@ -101,7 +101,8 @@ private[johnsnowlabs] class Bart(
       task: String,
       randomSeed: Option[Long] = None,
       ignoreTokenIds: Array[Int] = Array(),
-      beamSize: Int): Seq[Annotation] = {
+      beamSize: Int,
+      maxInputLength: Int): Seq[Annotation] = {
 
     val batchDecoder = sentences.grouped(batchSize).toArray.flatMap { batch =>
       val batchSP = encode(batch, task)
@@ -117,7 +118,8 @@ private[johnsnowlabs] class Bart(
         noRepeatNgramSize,
         randomSeed,
         ignoreTokenIds,
-        beamSize)
+        beamSize,
+        maxInputLength)
 
       decode(spIds)
 
@@ -176,10 +178,12 @@ private[johnsnowlabs] class Bart(
       noRepeatNgramSize: Int,
       randomSeed: Option[Long],
       ignoreTokenIds: Array[Int] = Array(),
-      beamSize: Int): Array[Array[Int]] = {
+      beamSize: Int,
+      maxInputLength: Int): Array[Array[Int]] = {
 
     val ignoreTokenIdsInt = ignoreTokenIds
-    val expandedEncoderInputIdsVals = batch.flatMap(x => List.fill(beamSize)(x))
+    val expandedEncoderInputIdsVals =
+      batch.flatMap(x => List.fill(beamSize)(x.take(maxInputLength)))
     val sequencesLength = expandedEncoderInputIdsVals.map(x => x.length).toArray
     val maxSentenceLength = sequencesLength.max // - curLen
 
@@ -492,6 +496,7 @@ private[johnsnowlabs] class Bart(
       noRepeatNgramSize = 0,
       randomSeed = Option(0),
       ignoreTokenIds = Array(0),
-      beamSize = 1)
+      beamSize = 1,
+      maxInputLength = 512)
   }
 }

--- a/src/main/scala/com/johnsnowlabs/nlp/annotators/seq2seq/BartTransformer.scala
+++ b/src/main/scala/com/johnsnowlabs/nlp/annotators/seq2seq/BartTransformer.scala
@@ -199,6 +199,18 @@ class BartTransformer(override val uid: String)
     this
   }
 
+  /** max length of the input sequence (Default: `0`)
+    *
+    * @group param
+    */
+  val maxInputLength =
+    new IntParam(this, "maxInputLength", "Maximum length of the input sequence")
+
+  def setMaxInputLength(value: Int): BartTransformer.this.type = {
+    set(maxInputLength, value)
+    this
+  }
+
   /** @group getParam */
   def getMinOutputLength: Int = $(this.minOutputLength)
 
@@ -477,6 +489,7 @@ class BartTransformer(override val uid: String)
     ignoreTokenIds -> Array(),
     batchSize -> 1,
     beamSize -> 4,
+    maxInputLength -> 512,
     useCache -> true)
 
   override def batchAnnotate(batchedAnnotations: Seq[Array[Annotation]]): Seq[Seq[Annotation]] = {
@@ -503,7 +516,8 @@ class BartTransformer(override val uid: String)
         task = $(task),
         randomSeed = this.randomSeed,
         ignoreTokenIds = $(ignoreTokenIds),
-        beamSize = $(beamSize))
+        beamSize = $(beamSize),
+        maxInputLength = $(maxInputLength))
     } else {
       Seq()
     }

--- a/src/test/scala/com/johnsnowlabs/nlp/annotators/seq2seq/BartTestSpec.scala
+++ b/src/test/scala/com/johnsnowlabs/nlp/annotators/seq2seq/BartTestSpec.scala
@@ -56,7 +56,7 @@ class BartTestSpec extends AnyFlatSpec {
       .show(truncate = false)
 
   }
-  "distilbart_xsum_12_6" should "handle text inputs longer than 512 and not crash" taggedAs FastTest in {
+  "distilbart_xsum_12_6" should "handle text inputs longer than 512 and not crash" taggedAs SlowTest in {
     // text longer than 512
     val testData = ResourceHelper.spark
       .createDataFrame(

--- a/src/test/scala/com/johnsnowlabs/nlp/annotators/seq2seq/BartTestSpec.scala
+++ b/src/test/scala/com/johnsnowlabs/nlp/annotators/seq2seq/BartTestSpec.scala
@@ -18,7 +18,7 @@ package com.johnsnowlabs.nlp.annotators.seq2seq
 
 import com.johnsnowlabs.nlp.base.DocumentAssembler
 import com.johnsnowlabs.nlp.util.io.ResourceHelper
-import com.johnsnowlabs.tags.SlowTest
+import com.johnsnowlabs.tags.{SlowTest, FastTest}
 import com.johnsnowlabs.util.Benchmark
 import org.apache.spark.ml.Pipeline
 import org.scalatest.flatspec.AnyFlatSpec
@@ -56,6 +56,77 @@ class BartTestSpec extends AnyFlatSpec {
       .show(truncate = false)
 
   }
+  "distilbart_xsum_12_6" should "handle text inputs longer than 512 and not crash" taggedAs FastTest in {
+    // text longer than 512
+    val testData = ResourceHelper.spark
+      .createDataFrame(
+        Seq(
+          (
+            1,
+            "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow." +
+              "PG&E stated it scheduled the blackouts in response to forecasts for high winds " +
+              "amid dry conditions. The aim is to reduce the risk of wildfires. Nearly 800 thousand customers were " +
+              "scheduled to be affected by the shutoffs which were expected to last through at least midday tomorrow.")))
+      .toDF("id", "text")
+
+    val documentAssembler = new DocumentAssembler()
+      .setInputCol("text")
+      .setOutputCol("documents")
+
+    val bart = BartTransformer
+      .pretrained("distilbart_xsum_12_6")
+      .setTask("summarize:")
+      .setInputCols(Array("documents"))
+      .setDoSample(true)
+      .setMaxOutputLength(30)
+      .setOutputCol("generation")
+
+    new Pipeline()
+      .setStages(Array(documentAssembler, bart))
+      .fit(testData)
+      .transform(testData)
+      .select("generation.result")
+      .show(truncate = false)
+  }
+
   "bart-large-cnn" should "run SparkNLP pipeline with maxLength=130 and doSample=true" taggedAs SlowTest in {
     val testData = ResourceHelper.spark
       .createDataFrame(

